src/main/java/pl/psnc/dei/ui/components/batches/BatchImportComponent.java

```diff
@@ -0,0 +1,211 @@
+        Predicate<Aggregator> predicate = agg -> agg.equals(Aggregator.EUROPEANA);
+        aggregatorsSelect = CommonComponentsFactory.getAggregatorSelector(predicate);
+        aggregatorsSelect.setEnabled(false);
+        aggregatorsSelect.setValue(Aggregator.EUROPEANA);
+        return aggregatorsSelect;
+    }
+
+    private Component createProjectSelection() {
+        projectsSelect = CommonComponentsFactory.getProjectSelector(projectsRepository);
+        projectsSelect.addValueChangeListener(event -> {
+            project = event.getValue();
+            datasetsSelect.setItems(batchService.getProjectDataset(project));
+        });
+        return projectsSelect;
+    }
+
+    private Component createDatasetsSelection() {
+        datasetsSelect = CommonComponentsFactory.getDatasetSelector(Collections.emptyList());
+        datasetsSelect.addValueChangeListener(event -> dataset = event.getValue());
+        return datasetsSelect;
+    }
+
+    private Component createImportNameInputField() {
+        importNameTextField = new TextField();
+        importNameTextField.addClassName("flex-1");
+        return importNameTextField;
+    }
+
+    private TextArea prepareRecordsTextArea() {
+        recordsTextArea = new TextArea();
+        recordsTextArea.setMinHeight("150px");
+        recordsTextArea.setWidthFull();
+        recordsTextArea.addClassName("query-form");
+        return recordsTextArea;
+    }
+
+    private void sendImport() {
+        if (!areInputsValid()) {
+            return;
+        }
+        Set<String> retrievedRecords;
+        try {
+            retrievedRecords = retrieveRecords();
+        } catch (ParseRecordsException ex) {
+            showNotification(ex.getMessage());
+            return;
+        }
+        try {
+            String datasetId = dataset != null ? dataset.getDatasetId() : null;
+            Set<Record> uploadedRecords = batchService.uploadRecords(project.getName(), datasetId, retrievedRecords);
+            if (uploadedRecords.isEmpty()) {
+                showNotification("There are no records to upload!");
+                return;
+            }
+            Import anImport = importPackageService.createImport(importNameTextField.getValue(), project.getProjectId(), uploadedRecords);
+            showNotification("Successfully created import: " + anImport.getName());
+        } catch (NotFoundException e) {
+            showNotification("Invalid input. " + e.getMessage());
+        }
+    }
+
+    private Set<String> retrieveRecords() throws ParseRecordsException {
+        String rawInput = recordsTextArea.getValue();
+        ObjectMapper mapper = new ObjectMapper();
+        Set<String> parsedRecords;
+        try {
+            parsedRecords = mapper.readValue(rawInput, new TypeReference<Set<String>>() {
+            });
+        } catch (JsonParseException | JsonMappingException jsonProcessingException) {
+            parsedRecords = readLines(rawInput);
```

Comment:
I am not sure if this is actually enough, because for the input like this:

[
/12325/acc,
/12344/fff
]

which is not the correct JSON array the notification says "Invalid record [". Maybe some better parsing should be involved here.

On the other hand when the valid JSON is supplied:

[
"/12325/acc",
"/12344/fff"
]

there is no error at all although both identifiers do not exits in Europeana and the log shows 404 error.

Reply:
Improved parsing error handling